### PR TITLE
LPS-161562 Break the circular dependency between CPDefinitionLocalSer…

### PR DESCRIPTION
**Pull request by slnn**
https://issues.liferay.com/browse/LPS-161562

Hi @tinatian @dantewang 

This pull breaks the circular dependency between CPDefinitionLocalService and CProductLocalService by using persistence class directly.

Lily

### CODE REVIEW

**Comment by slnn** [comment]
Hi @zhangyan0701 

Could you help to scan test results when they are finished?

Thanks
Lily

**Comment by zhangyan0701** [comment]
ci:test:commerce

**Comment by zhangyan0701** [comment]
No regressions found.

**Review by dantewang — APPROVED** [review]

**Comment by slnn** [comment]
Hi @dantewang 

Since the change for disabling auto import ([LPS-161869](https://issues.liferay.com/browse/LPS-161869)) has merged into master, I rebase this branch with the master and forced push on this pull.

Lily

**Comment by slnn** [comment]
With the forced push change, the ci tests has been ran in:https://github.com/slnn/liferay-portal/pull/628

**Comment by tinatian** [comment]
@shuyangzhou This pull is good to go.

Tina

**Comment by tinatian** [comment]
@slnn Please remove deleteCPDefinitions(long cProductId, int status) as no one is using it now. Please also send a message to t-commerce slack group about the review of this pull.

Thanks.
Tina.

**Comment by slnn** [comment]
Hi @tinatian 

I have removed the method deleteCPDefinitions(long cProductId, int status) and forced pushed this pull and  send a message to t-commerce slack group for review this pull.

Lily

**Comment by victoroscuro** [comment]
ci:test:commerce

File: modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CProductLocalServiceImpl.java [hunk 1]
```diff
@@ -74,7 +74,7 @@ public CProduct deleteCProduct(CProduct cProduct) throws PortalException {
 
 		// Commerce product definitions
 
-		_cpDefinitionLocalService.deleteCPDefinitions(
+		_cpDefinitionPersistence.removeByC_S(
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CProductLocalServiceImpl.java:77]: this I'm not sure will work, with the service call we delete all the related entities of the prodct, this way only the DB entities will be removed.

Comment (dantewang) [reply modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CProductLocalServiceImpl.java:77]: @andrea-ale-sbarra Is that PR going to be merged soon?

Comment (dantewang) [reply modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CProductLocalServiceImpl.java:77]: @andrea-ale-sbarra Ah I just see the meesage in the slack channel, thanks!

Comment (slnn) [reply modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CProductLocalServiceImpl.java:77]: Hi @andrea-ale-sbarra 

The commit https://github.com/liferay-commerce/liferay-portal/pull/2997/commits/769758030a90ad035215fc49869e94417192eab3 of the pull:
https://github.com/liferay-commerce/liferay-portal/pull/2997 has removed all the usage of CPDefinitionLocalService in CProductLocalServiceImpl, but still kept the reference of CPDefinitionLocalService in it, why should we keep it?

@dantewang @tinatian  if the reference of CPDefinitionLocalService is removed in CProductLocalServiceImpl in the [COMMERCE](https://github.com/liferay-commerce/liferay-portal/pull/2997)  change, then we do not need to do the change of LPS-161562 in this pull.

Lily

**Comment by victoroscuro** [comment]
ci:test:commerce

**Comment by tinatian** [comment]
ci:forward

**Comment by tinatian** [comment]
FYI @shuyangzhou

**Comment by tinatian** [comment]
@slnn This needs a rebase

**Pull request closed without merging**